### PR TITLE
feat: device flow

### PR DESCRIPTION
**Pull request by juguhan**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: device_authorize_write.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	rw.Header().Set("Cache-Control", "no-store")
+	rw.Header().Set("Pragma", "no-cache")
+
+	deviceResponse := &DeviceAuthorizationResponse{
+		DeviceCode:              responder.GetDeviceCode(),
+		UserCode:                responder.GetUserCode(),
+		VerificationURI:         responder.GetVerificationURI(),
+		VerificationURIComplete: responder.GetVerificationURIComplete(),
+		ExpiresIn:               responder.GetExpiresIn(),
+		Interval:                responder.GetInterval(),
```
Comment (asasmita) [inline device_authorize_write.go]: how about the Extra? It is not being output here?

File: handler/rfc8628/device_user_verification_handler.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+type DeviceUserVerificationHandler struct {
+	Storage  RFC8628CodeStorage
+	Strategy RFC8628CodeStrategy
+	Config   interface {
+		fosite.DeviceAuthorizationProvider
+		fosite.DeviceAndUserCodeLifespanProvider
+	}
+}
+
```
Comment (asasmita) [inline handler/rfc8628/device_user_verification_handler.go]: can consider add
`var _ fosite.DeviceUserVerificationEndpointHandler = (* DeviceUserVerificationHandler)(nil)`

File: handler/openid/flow_device_authorization.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	OpenIDConnectRequestValidator *OpenIDConnectRequestValidator
+	oauth2.CodeTokenEndpointHandler
+
+	Config interface {
+		fosite.IDTokenLifespanProvider
+	}
+
+	*IDTokenHandleHelper
+}
+
```
Comment (asasmita) [inline handler/openid/flow_device_authorization.go:26]: can consider add
`var _ fosite.DeviceUserVerificationEndpointHandler = (* OpenIDConnectDeviceAuthorizationHandler)(nil)`
`var _ fosite.TokenEndpointHandler = (* OpenIDConnectDeviceAuthorizationHandler)(nil)`

**Review by asasmita — COMMENTED** [review]
I have some comments

File: device_authorize_write.go (hunk 1 above)
Comment (juguhan) [reply device_authorize_write.go]: I'm not sure how we want to use extra here.

**Review by asasmita — DISMISSED** [review]
Looks good to me

**Review by vivshankar — DISMISSED** [review]

**Review by vivshankar — APPROVED** [review]

**Pull request merged**